engine/access/state_stream/backend/handler.go

```diff
@@ -215,3 +217,260 @@ func (h *Handler) GetRegisterValues(_ context.Context, request *executiondata.Ge
 
 	return &executiondata.GetRegisterValuesResponse{Values: values}, nil
 }
+
+func (h *Handler) createAccountStatusesFilter(statusFilter *executiondata.StatusFilter) (state_stream.EventFilter, error) {
```

Comment:
this doesn't quite capture the desired behavior.

1. No Event Types / No Addresses
  -> match on all events in the default list without inspecting the payloads
2. Some Events / No Addresses
  -> match on the provided list of events without inspecting the payloads
3. No Event Types / Some Addresses
  -> match on any event from the default list that matches an address in the list. must inspect the payloads
4. Some Events / Some Addresses
  -> match on the provided list of events that matches an address in the list. must inspect the payloads

All event type provided by the user must be in the default list of events.
All addresses must be validated using the event filter validator.

If no `statusFilter` is provided, we still need to build an `EventFilter` with the above criteria, otherwise it will match all events.